ipyparallel/client/asyncresult.py

```diff
@@ -362,7 +367,17 @@ def __getattr__(self, key):
         except (error.TimeoutError, KeyError):
             raise AttributeError("%r object has no attribute %r"%(
                     self.__class__.__name__, key))
-
+    
+    @staticmethod
+    def _wait_for_child(child, timeout=_FOREVER, evt=None):
+        """Wait for a child to be done"""
+        if evt is not None:
+            evt = Event()
+        else:
+            evt.unset()
+        child.add_done_callback(lambda f: evt.set())
```

Comment:
Also looking at the usage, it does not seem like you ever let a change for `evt` to be `None`.